edk2toolext/environment/repo_resolver.py

```diff
@@ -418,6 +418,22 @@ def clean(abs_file_system_path, ignore_files=[]):
         repo.git.clean(*params)
 
 
+def submodule_clean(abs_file_system_path, submodule, ignore_files=None):
+    """Resets and cleans a submodule of the repo.
+
+    Args:
+        abs_file_system_path (PathLike): repo directory
+        submodule (obj): object containing path (relative) attribute
+        ignore_files (list, optional): list of files to ignore when performing a clean. Defaults to [].
+    """
+    submodule_path = Path(submodule.path).as_posix()
```

Comment:
I don't think you need to do `.as_posix()`

either do `str(Path(submodule.path))` or just pass submodule.path directly. Either looks like it works for me.

Reply:
When the submodule path has backslashes, I was seeing an error:
```pdb
(Pdb) repo.submodule('Common/MU')
git.Submodule(name=Common/MU, path=Common/MU, url=https://github.com/Microsoft/mu_plus, branch_path=refs/heads/release/202208)
(Pdb) repo.submodule('Common\\MU')
*** ValueError: Didn't find submodule named 'Common\\MU'
(Pdb) l
433             (ValueError): submodule's path was invalid
434         """
435         submodule_path = Path(submodule.path).as_posix()
436         with Repo(abs_file_system_path) as repo:
437             breakpoint()
438  ->         if repo.submodule(submodule_path).module_exists():
439                 clean(os.path.join(abs_file_system_path, submodule.path), ignore_files or [])
440
441
442     def submodule_resolve(abs_file_system_path, submodule, omnicache_path=None):
443         """Resolves a submodule to the specified branch and commit in .gitmodules.
(Pdb)
```